vk_video_decoder/libs/NvVideoParser/src/VulkanH264Parser.cpp

```diff
@@ -2353,7 +2353,7 @@ bool VulkanH264Decoder::slice_header(slice_header_s *slh, int nal_ref_idc, int n
             {
                 slh->store_ref_base_pic_flag = u(1);
                 if ((slh->nhe.svc.use_ref_base_pic_flag || slh->store_ref_base_pic_flag) && !slh->IdrPicFlag)
-                    slh->adaptive_ref_pic_marking_mode_flag = (unsigned char)dec_ref_base_pic_marking(slh->mmbco);
+                    slh->adaptive_ref_pic_marking_mode_flag = (uint32_t)dec_ref_base_pic_marking(slh->mmbco) & 0x1;
```

Comment:
This should be:
slh->adaptive_ref_pic_marking_mode_flag = (dec_ref_base_pic_marking(slh->mmbco) != 0);